From 0d4b65082e2feba9818950c9d4fc2facdac13177 Mon Sep 17 00:00:00 2001
From: George Zacharia <george.zcharia@gmail.com>
Date: Tue, 19 Jan 2021 22:33:56 +0530
Subject: [PATCH] LatinIME: Remove hardcoded emoji row logic

The hardcoded value was not allowing adjusting the actionbar height, make this follow the overalyable row height

The value is incremented by one as the actionbar is considered one row.

Change-Id: Ia3135bf5a3ef38224ce4e7dadc812faf9efeb69d
---
 .../inputmethod/keyboard/emoji/EmojiLayoutParams.java       | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/java/src/com/android/inputmethod/keyboard/emoji/EmojiLayoutParams.java b/java/src/com/android/inputmethod/keyboard/emoji/EmojiLayoutParams.java
index 797541a15..d3d050e3e 100644
--- a/java/src/com/android/inputmethod/keyboard/emoji/EmojiLayoutParams.java
+++ b/java/src/com/android/inputmethod/keyboard/emoji/EmojiLayoutParams.java
@@ -25,8 +25,6 @@ import com.android.inputmethod.latin.R;
 import com.android.inputmethod.latin.utils.ResourceUtils;
 
 final class EmojiLayoutParams {
-    private static final int DEFAULT_KEYBOARD_ROWS = 4;
-
     public final int mEmojiPagerHeight;
     private final int mEmojiPagerBottomMargin;
     public final int mEmojiKeyboardHeight;
@@ -52,7 +50,9 @@ final class EmojiLayoutParams {
                 (int) (res.getDimension(R.dimen.config_emoji_category_page_id_height));
         final int baseheight = defaultKeyboardHeight - mBottomPadding - mTopPadding
                 + mKeyVerticalGap;
-        mEmojiActionBarHeight = baseheight / DEFAULT_KEYBOARD_ROWS
+        final int mNumberOfRows = (int) (1.0f / res.getFraction(
+                R.fraction.config_emoji_keyboard_row_height, 1, 1)) + 1;
+        mEmojiActionBarHeight = baseheight / mNumberOfRows
                 - (mKeyVerticalGap - mBottomPadding) / 2;
         mEmojiPagerHeight = defaultKeyboardHeight - mEmojiActionBarHeight
                 - mEmojiCategoryPageIdViewHeight;
